Project/PrivateVault App/Code/Model/Extensions/Folder+Creation.swift

```diff
@@ -0,0 +1,26 @@
+//
+//  Folder+Creation.swift
+//  PrivateVault
+//
+//  Created by Elena Meneghini on 06/08/2021.
+//
+
+import CoreData
+
+extension Folder {
+	convenience init(context: NSManagedObjectContext, name: String) {
+		self.init(context: context)
+		self.name = name
+	}
+}
+
+extension Folder {
+	var children: [Folder]? {
```

Comment:
I avoid optional arrays because it's not always clear if there should be a difference between `nil` and an empty array.
This could look like this:

```
var children: [Folder] {
  subfolders?.allObjects as? [Folder] ?? []
}
```

Reply:
I'm kind of forced to use an optional array because of the List requirements: you need an optional array of children to get the expandable arrow. I also need to check if `array?.isEmpty == true` because If I return just `subfolders?.allObjects as? [Folder]` for example, then in `FolderSelectionView` I will get the expandable arrow for all folders included the ones without children.